lib/ngx/req.lua

```diff
@@ -0,0 +1,54 @@
+-- Copyright (C) Yichun Zhang (agentzh)
+
+
+local ffi = require 'ffi'
+local base = require "resty.core.base"
+
+
+local FFI_BAD_CONTEXT = base.FFI_BAD_CONTEXT
+local C = ffi.C
+local ffi_str = ffi.string
+local get_string_buf = base.get_string_buf
+local get_size_ptr = base.get_size_ptr
+local getfenv = getfenv
+local error = error
+local tonumber = tonumber
+
+
+local DEFAULT_EXT_MAXSIZE = 255
+
+
+local _M = { version = base.version }
+
+
+ffi.cdef[[
+    int ngx_http_lua_ffi_req_get_ext(ngx_http_request_t *r, char *out,
+        size_t *len);
+]]
+
+
+function _M.get_uri_ext(ext_len)
+    local r = getfenv(0).__ngx_req
+    if not r then
+        return error("no request found")
+    end
+
+    ext_len = tonumber(ext_len) or DEFAULT_EXT_MAXSIZE
+    if ext_len <= 0 then
```

Comment:
nitpick: our recent idiom across the Lua-land codebases is to use branch-free algorithms like `math.max()`/`math.min()` to handle these scenarios.